torch_xla/core/xla_model.py

```diff
@@ -974,6 +976,56 @@ def wait_device_ops(devices=[]):
   torch_xla._XLAC._xla_wait_device_ops(devices=devices)
 
 
+def bucketed_allreduce(gradients):
+    total = 0
+    tensor_bucket = []
+
+    for grad in gradients:
+      grad_bytes = grad.numel() * grad.element_size()
+
+      # Gradient is larger than bucket_cap, don't bucketize
+      if grad_bytes > bucket_cap:
```

Comment:
Curious why you want to specialize this case?

Reply:
if the grad_bytes (already in the tensor) is larger than bucket cap, we send it straight away as a single tensor instead of bucketing.

Reply:
Right, I understood the logic. But why? Combining it with the bucket introduce some problems?

Reply:
Yeah, looks like you can get rid of this if statement (until continue), and the "if total > bucket_cap" should take care of this condition when total=0.

Reply:
The issue with combining this with the rest is that the "buffer" allocated in the underlying runtime may not have enough space to fit this large tensor. The idea is to have a large buffer that can fit all the tensors. It can happen that total_bytes is just below the max allowed and this tensor if added to the bucket spills the maximum. Hence should go "alone" without bucketizing

Reply:
See your concerns now !! Fixed the code flow